add validation to capa problem authoring markdown

HamzaIbnFarooq opened the pull request:
### Related Issue
https://github.com/mitodl/edx-platform/pull/239
https://github.com/mitodl/edx-platform/pull/240

### What this PR does
It adds validation to CAPA problems and shows a notification to the course author regarding the issue, along with fixing old response report generation issues by providing a default value for the problems having no text values for answer options.

### How to test this PR
To understand the purpose of this PR, follow the below steps to reproduce the issue:

1) Create a Checkbox Problem on studio
2) Don't put any text against one of the options
![image](https://user-images.githubusercontent.com/42243411/107644036-b906ee00-6c98-11eb-9616-7d064a2871f3.png)
3) Through LMS select that option with no text
4) Go to `Data Download` under `Instructor` tab
5) Under the `Reports` section, select the Problem and press `Create a report of problem responses`
The error will be generated.

Now use the current PR and go through the same process, and you will see that the editor will not allow saving such content and mentions an error message instead.


### Screenshots
![image](https://user-images.githubusercontent.com/42243411/108207094-198aa500-7149-11eb-8642-242f14316d09.png)

Commit:
fix: add validation to capa problem markdown for missing option text

Before this commit, if a course author created a capa mcqs or similar problem without providing any answer text for an option the question would be created causing abnormal behavior for learners. This commit will validate answer text of all options and raise an error message to author to fix the issue on the go.

ziafazal commented on common/lib/xmodule/xmodule/js/src/problem/edit.js:
```diff
@@ -402,6 +402,9 @@
                         if (line.length > 0) {
                             textHint = extractHint(line, true);
+                            if (!textHint.nothint) {
+                                throw new Error("One of the provided options don't have valid text value")
```
replace `don't have valid` with `doesn't have a valid`. Are these error messages are marked for internationalization?

HamzaIbnFarooq replied:
I have fixed the typo and the messages are already internationalized in the `showErrorMeassage` method but I have added translations to individual strings as well.

ziafazal reviewed (commented):
@HamzaIbnFarooq overall LGTM. few typos needs to be fixed and I have question.

ziafazal reviewed (approved):
👍

HamzaIbnFarooq committed:
fix: adds default texts for missing answer texts in capa problems to fix response report generation

If an author has created a capa problem like an mcqs or something similar without providing answer text to an option and some learner selected that option then the response report generation will fail due to that missing answer text. The current commit will add default text to be substituted and prevents report generation crash.

HamzaIbnFarooq commented:
@ziafazal I merged https://github.com/mitodl/edx-platform/pull/264 into the current PR and added some tests as well. Kindly have a final look before I create an upstream PR.

ziafazal reviewed (approved):
👍

HamzaIbnFarooq commented:
@pdpinch I have created an upstream PR related to this feature: https://github.com/edx/edx-platform/pull/27858

Commit:
refactor: refactoring error messages for CAPA problem authoring

HamzaIbnFarooq commented:
The upstream PR https://github.com/edx/edx-platform/pull/27858 has been merged so closing this internal PR as well
